adr/0006-dev-discussions.adoc

```diff
@@ -0,0 +1,125 @@
+* Allow use of e-mail as not everyone can/will use web-ui
+* Have consistent place to publish and lookup Quarkus development ideas/suggestions
+
+== Scenarios 
+
+For a Quarkus core developer, where do they best post ideas for suggestion and feedback? Zulip chat? Open an issue? Submit a PR directly? Post to quarkus-dev? Open a GitHub discussion? Use private chat? Today all of the above happens based on habit or specific requests.
+
+A dev interested in doing contributions by following the ongoing conversations - which of the many channels do they follow? 
+
+== Considered options 
+
+Primarily use quarkus-dev mailing list::
+    Advantages::::
+        * well-known and understood by most if not all existing Quarkus contributors (~1000+ subscribers)
+        * can post, comment and read with any e-mail client
+        * filters out most noise as people need to subscribe
+    Disadvantages::::
+        * over time it has died out, rarely where conversations starts/continues being discussed
+        * e-mail/google groups is just not where new people show up as much now
+        * requires subscribing to specific list to post
+        * spam (fixed now by moderation)
+        * arbitrary blocking of mails (specifally @gsmet and others)
+
+Primarily have "Design Discussions" a sub-category of Quarkus Community/User Discussions::
+    Advantages::::
+        * Centralized location for discussions, anyone sees it
+        * Anyone can read
+        * Anyone with GitHub account can post/reply
+        * Moderation tools available to battle spam/noise
+    Disadvantages::::
+        * Notifications via email comes for *all* discussions, not possible to choose specific category
+        * User forums gets intermixed with Developer/Design discussions
+
+Primarily use issues/PR's::
+    Advantages::::
+        * Conversation directly linked to code changes
+    Disadvantages::::
+        * Not possible to just monitor the overall design discussions as all mixed up with anything else
+        * Notification filtering is quite limited
+        * Limited threading in webui
+        * No-threading on email responses
+
+Primarily use Zulip Chat::
+    Advantages::::
+        * Real-time communication
+    Disadvantages::::
+        * Only for those who have time/privilege to follow along
+
+Use private chat/mail::
+    Advantages::::
+        * Direct and focused private communication (sometimes useful)
+    Disadvantages::::
+        * Not great for open governance unless culture to comment/report on conclusion somewhere more public
+        * Cannot be located/found when trying to search for answers/why something was done
+
+Status quo/Do nothing::
+    Advantages::::
+        * Nothing need to change
+    Disadvanteges::::
+        * All of the above disadvantages
+        * No clear place to go for design discussions
+
+Seperate quarkusio/quarkus-dev repository with discussions::
```

Comment:
I have a suggestion: What if we called it `quarkus-contrib`? We see confusion with both the `quarkus-dev` google group and the `#dev` zulip channel, because _every single one_ of our users is a developer, by definition. So people think "Awesome! I'm a Quarkus developer, this is the place for me!", and then we have to spend effort re-routing their queries. Using `-contrib` avoids this ambiguity.

Reply:
I considered it but I when away from it as:

A) quarkus-contributors gets to be quite long and noisy in mail subject
B) "contrib" in many projects is additional stuff outside the main development 
C) If worried about those used to quarkus-dev ml not discovering it reusing the name helps locate it via search. 

WDYT ? Valid arguments ? Asking as I could be swayed both ways. B is my biggest worry but might just be from early days :)

Reply:
Some comments
on A : `[quarkus/quarkus-dev]` will already make it not very usable on mobile emails leaving 10 characters before it hides the rest of the email title.
On C, my personal importance meter rates this not too high

B is real and it adds up with GitHub discovery path difficulties to _Discussions_.

Reply:
> B is real and it adds up with GitHub discovery path difficulties to _Discussions_.

not sure I follow ? I read it as yes, its problematic to use "-contrib" as it has a specific meaning but then i missed your point on how that connects to discovery path difficulties to discussions?

Reply:
That's a good point about `-contrib` being used for 'extra' stuff.  Using `contrib` for that kind of content is a silly name, IMO, but it's out there. So if the aim is to avoid ambiguity, then `contrib` is no good. So then `contributors` is my best suggestion, but I agree it's long. That's maybe not fatal. Another alternative is `quarkus-design`. Or `quarkus-arch` or `quarkus-architecture`, although that's a bit less differentiated from users, IMO. 

I think @emmanuelbernard's point is that the discovery experience for the new repo won't be great. It won't be the first place people will expect to find design discussions, they're going to need to be directed to it. If the name of the repo looks like it holds additional stuff outside the main development, then it's even harder for people to land in the right place when they want to talk design.